packages/office-ui-fabric-react/src/components/Callout/CalloutContent.base.tsx

```diff
@@ -122,8 +130,11 @@ export class CalloutContentBase extends BaseComponent<ICalloutProps, ICalloutSta
       this._setHeightOffsetEveryFrame();
     }
 
-    // if the callout becomes hidden, then remove any positions, bounds that were placed on it.
-    if (newProps.hidden && newProps.hidden !== this.props.hidden) {
+    // Ensure positioning is recalculated when we are about to show a persisted menu.
+    if (!newProps.hidden && newProps.hidden !== this.props.hidden) {
```

Comment:
Isn't `!newProps.hidden && newProps.hidden` always going to result in `false`?

Reply:
!newProps.hidden is evaluated first, then newProps.hidden !== this.props.hidden, so this is, if the new value is not hidden, and the value is changing which in this case means from not hidden to hidden, run the code

Reply:
I think you should add another set of parenthesis to make that a bit more clear.